PurchasesCoreSwiftTests/Purchasing/ProductInfoTests.swift

```diff
@@ -88,63 +88,62 @@ class ProductInfoTests: XCTestCase {
         let productInfo: ProductInfo = .createMockProductInfo(subscriptionGroup: subscriptionGroup)
         expect(productInfo.asDictionary()["subscription_group_id"] as? String) == subscriptionGroup
     }
-
+    
     func testAsDictionaryConvertsDiscountsCorrectly() {
-        if #available(macOS 10.14.4, *) {
-
-
-            let discount1 = PromotionalOffer(withProductDiscount: MockProductDiscount(identifier: "offerid1",
-                                                                                      paymentMode: .payAsYouGo,
-                                                                                      price: 11))
-            let discount2 = PromotionalOffer(withProductDiscount: MockProductDiscount(identifier: "offerid2",
-                                                                                      paymentMode: .payUpFront,
-                                                                                      price: 12))
-            let discount3 = PromotionalOffer(withProductDiscount: MockProductDiscount(identifier: "offerid3",
-                                                                                      paymentMode: .freeTrial,
-                                                                                      price: 13))
-
-            let productInfo: ProductInfo = .createMockProductInfo(discounts: [discount1, discount2, discount3])
-
-            expect(productInfo.asDictionary()["offers"] as? [[String: NSObject]]).toNot(beNil())
-            guard let receivedOffers = productInfo.asDictionary()["offers"] as? [[String: NSObject]] else { fatalError() }
-
-            expect(receivedOffers[0]["offer_identifier"] as? String) == discount1.offerIdentifier
-            expect(receivedOffers[0]["price"] as? NSDecimalNumber) == discount1.price
-            expect((receivedOffers[0]["payment_mode"] as? NSNumber)?.intValue) == discount1.paymentMode.rawValue
-
-            expect(receivedOffers[1]["offer_identifier"] as? String) == discount2.offerIdentifier
-            expect(receivedOffers[1]["price"] as? NSDecimalNumber) == discount2.price
-            expect((receivedOffers[1]["payment_mode"] as? NSNumber)?.intValue) == discount2.paymentMode.rawValue
-
-            expect(receivedOffers[2]["offer_identifier"] as? String) == discount3.offerIdentifier
-            expect(receivedOffers[2]["price"] as? NSDecimalNumber) == discount3.price
-            expect((receivedOffers[2]["payment_mode"] as? NSNumber)?.intValue) == discount3.paymentMode.rawValue
-        }
+        let discount1 = PromotionalOffer(offerIdentifier: "offerid1",
```

Comment:
the test is the same, I've only changed how the discounts are initialized. The constructor with a product discount as param is not available for all OS versions, but there is another compatible one. Since the objective of this test case is not to test if the constructor works fine, I think we can use the most compatible constructor. WDYT?